estate/models/estate_property.py

```diff
@@ -0,0 +1,111 @@
+from odoo import api,models, fields
+from datetime import date
+from dateutil.relativedelta import relativedelta
+from odoo.exceptions import UserError,ValidationError
+from odoo.tools.float_utils import float_compare 
+
+class EstateProperty(models.Model):
+    _name = "estate.property" 
+    _description = "Real Estate Property"
+
+    name = fields.Char("Estate Name", required=True,)
+    description = fields.Text("Description",)
+    postcode = fields.Char("Postalcode")
+    date_availability = fields.Date('Date Availability', default=lambda self: date.today() + relativedelta(months=3))
+    expected_price = fields.Float('Expected Price', required=True )
+    selling_price = fields.Float('Selling Price', readonly=True)
+    bedrooms = fields.Integer("Bedrooms", default=2)
+    living_area = fields.Integer("Living Area (sqm)")
+    facades = fields.Boolean("Facades")
+    garage = fields.Boolean("Garage")
+    garden = fields.Boolean("Garden")
+    garden_area = fields.Integer("Garden Area (sqm)")
+    garden_orientation = fields.Selection( selection = [
+        ('north','NORTH'),
+        ('south','SOUTH'),
+        ('west','WEST'),
+        ('east','EAST'),
+        ])
+    active = fields.Boolean("Active", default=True)
+    state = fields.Selection(selection = [
+        ('new','NEW'),
+        ('offer_received','OFFER RECEIVED'),
+        ('offer_accepted','OFFER ACCEPTED'),
+        ('sold','SOLD'),
+        ('cancelled','CANCELLED'),
+        ], default='new')
+    property_type_id = fields.Many2one('estate.property.type', string='Real Estate Type')
+    buyer_id = fields.Many2one('res.partner', string='Buyer',)
+    salesperson_id = fields.Many2one('res.partner', string='Salesperson', default=lambda self: self.env.user)
+    tag_ids = fields.Many2many('estate.property.tag', string='Real Estate Tag')
+    offer_ids = fields.One2many('estate.property.offer', inverse_name='property_id')
+    total_area = fields.Float('Total Area (sqm)', compute='_compute_total_area', readonly=True)
+    best_offer = fields.Float('Best Offer', compute='_compute_best_offer', readonly=True)
+    type_id = fields.Many2one('estate.property.type')
+    _order = 'id desc'
+
+    _sql_constraints = [
+        ('expected_price', 'CHECK(expected_price > 0)',
+         'The expected price should be strictly grater than 0.'),
+        ('selling_price', 'CHECK(selling_price > 0)',
+         'The selling price should be strictly grater than 0.'),
```

Comment:
Don't forget that the selling price might be 0 if no offer has been accepted yet